tests/integration/replication-4.tcl

```diff
@@ -54,30 +54,54 @@ start_server {tags {"repl external:skip"}} {
         test {With min-slaves-to-write (1,3): master should be writable} {
             $master config set min-slaves-max-lag 3
             $master config set min-slaves-to-write 1
-            $master set foo bar
-        } {OK}
+            assert_equal OK [$master set foo 123]
+            wait_for_condition 50 100 {
+               [$slave get foo] eq {123}
+            } else {
+                fail "Write did not synced to slave"
+            }
+            assert_equal OK [$master eval "return redis.call('set','foo',12345)" 0]
+            wait_for_condition 50 100 {
+               [$slave get foo] eq {12345}
+            } else {
+                fail "Write did not synced to slave"
+            }
+        }
 
         test {With min-slaves-to-write (2,3): master should not be writable} {
             $master config set min-slaves-max-lag 3
             $master config set min-slaves-to-write 2
-            catch {$master set foo bar} e
-            set e
-        } {NOREPLICAS*}
+            assert_error "*NOREPLICAS*" {$master set foo bar}
+            assert_error "*NOREPLICAS*" {$master eval "redis.call('set','foo','bar')" 0}
+        }
 
         test {With min-slaves-to-write: master not writable with lagged slave} {
             $master config set min-slaves-max-lag 2
             $master config set min-slaves-to-write 1
-            assert {[$master set foo bar] eq {OK}}
+            assert_equal OK [$master set foo 123]
+            assert_equal OK [$master eval "return redis.call('set','foo',12345)" 0]
+            # Killing a slave to make it become a lagged slave.
             exec kill -SIGSTOP [srv 0 pid]
+            # Waiting for slave kill.
             wait_for_condition 100 100 {
-                [catch {$master set foo bar}] != 0
+                [catch {$master set foo 123}] != 0 &&
+                [catch {$master eval "redis.call('set','foo',12345)" 0}] != 0
```

Comment:
i don't think we need that extra condition for the wait.
the wait will wait for the right state regardless, and the assertion below verifies both ways give the right error.